### DISCUSSION TITLE
The `this` keyword for lambda

### DISCUSSION THREAD

**Discussion by IgorSivirinov** [dotnet/csharplang#4946] [Language Ideas] [answered]
### Summary
I would like to suggest adding the ability to allow `this` to be specified as a lambda parameter.
That is, we say that in a given lambda `this` it is not only. class, but also a parameter. That is, all fields, methods, etc. class and lambda parameter are accessible with `this`
Example LINQ query:
```C#
users.Where(this => Age < 35).ToList();
```
Usage example with ReactiveUI DataBinding:
```C#
this.WhenActivated(this =>
{
    this.Bind(ViewModel, this => TheText, this => TheTextBox.Text)
        .DisposeWith(this);
    this.OneWayBind(ViewModel, this => TheText, this => TheTextBlock.Text)
        .DisposeWith(this);
    this.BindCommand(ViewModel, this => TheTextCommand, this => TheTextButton)
        .DisposeWith(this);
});
```

### Motivation
I would like to reduce the amount of writing the same parameter.

### Drawbacks
The main problem is the conflict with the class the lambda is in. I suggest if the variable, method, etc. in the class is called the same as in the lambda parameter object and we need to access the variable from the class, then we use this syntax
`this.out`

By default, in such a conflict, the variable from the lambda parameter is used.
If we need to get the value of a parameter, we use `this`

**Comment by jnm2** [comment]
An alternative approach to reduce writing a parameter that is even shorter: https://github.com/dotnet/csharplang/discussions/91

**Comment by HaloFour** [comment, marked as answer]
Also see https://github.com/dotnet/csharplang/discussions/1151 where I think replacing "this" makes more sense.
